tests/unit/sokoban_test.cpp

```diff
@@ -50,4 +50,83 @@ TEST_CASE("should not be solved when empty goal cell is present") {
     }};
     Sokoban soko(levels);
     CHECK(!soko.solved());
+}
+
+TEST_CASE("Should not mutate the board when move is invalid") {
```

Comment:
We should probably add more tests to make sure moving works in all directions, but these are a great start and you can follow the same format now so it should be pretty effortless to write cases.

The test board could even be a row smaller to save a bit of space, but minor point.